### ISSUE TITLE
'Name already defined' error when importing function from module with the same name

### ISSUE THREAD

**Issue by padix-key** [python/mypy#5479]
*mypy* shows an unexpected error, when I import all attributes from a module using `*` and a function has the same name as the module.

Example:
I have a package named `foo`, that contains the files `__init__.pyi`, `a.pyi`, `b.pyi` and `c.pyi`.

```
# a.pyi
def a() -> None: ...

# b.pyi
def not_b() -> None: ...

# c.pyi
def c() -> None: ...

# __init__.pyi
from .a import *
from .b import *
from .c import c
```

`mypy foo` shows the following error:
`foo/__init__.pyi:1: error: Name 'a' already defined (by an import)`.

I have used *mypy* 0.620.

**Comment by ilevkivskyi** [comment]
While trying to reproduce this, I have found mypy actually _crashes_ on master with this. A simple repro:
```python
from .whatever import smth
```
The traceback:
```
  File "/Users/ilevkivskyi/mypy_dev/bin/mypy", line 11, in <module>
    load_entry_point('mypy==0.630+dev.f91aae92e52c08d970bd6e99bdf0702a17032014', 'console_scripts', 'mypy')()
  File "/Users/ilevkivskyi/mypy_dev/lib/python3.6/site-packages/mypy/__main__.py", line 7, in console_entry
    main(None)
  File "/Users/ilevkivskyi/mypy_dev/lib/python3.6/site-packages/mypy/main.py", line 91, in main
    res = type_check_only(sources, bin_dir, options, flush_errors, fscache)  # noqa
  File "/Users/ilevkivskyi/mypy_dev/lib/python3.6/site-packages/mypy/main.py", line 148, in type_check_only
    fscache=fscache)
  File "/Users/ilevkivskyi/mypy_dev/lib/python3.6/site-packages/mypy/build.py", line 177, in build
    flush_errors, fscache)
  File "/Users/ilevkivskyi/mypy_dev/lib/python3.6/site-packages/mypy/build.py", line 350, in _build
    graph = dispatch(sources, manager)
  File "/Users/ilevkivskyi/mypy_dev/lib/python3.6/site-packages/mypy/build.py", line 2520, in dispatch
    graph = load_graph(sources, manager)
  File "/Users/ilevkivskyi/mypy_dev/lib/python3.6/site-packages/mypy/build.py", line 2715, in load_graph
    caller_state=st, caller_line=st.dep_line_map.get(dep, 1))
  File "/Users/ilevkivskyi/mypy_dev/lib/python3.6/site-packages/mypy/build.py", line 1854, in __init__
    ancestor_for, root_source, skip_diagnose=temporary)
  File "/Users/ilevkivskyi/mypy_dev/lib/python3.6/site-packages/mypy/build.py", line 2415, in find_module_and_diagnose
    if not (options.ignore_missing_imports or in_partial_package(id, manager)):
  File "/Users/ilevkivskyi/mypy_dev/lib/python3.6/site-packages/mypy/build.py", line 2439, in in_partial_package
    temporary=True)
  File "/Users/ilevkivskyi/mypy_dev/lib/python3.6/site-packages/mypy/build.py", line 1834, in __init__
    assert id or path or source is not None, "Neither id, path nor source given"
AssertionError: Neither id, path nor source given
```
It doesn't matter whether the module exists or not. My diagnosis is that when seeing `.whatever`, mypy splits it into two strings `''` and `'whatever'` and checks whether first one exists. In other places we use `correct_rel_imp`. Probably we can do something similar here.

**Comment by ilevkivskyi** [comment]
OK, the crash itself is a duplicate of https://github.com/python/mypy/issues/2974 so lowering priority to normal. I also remove the crash label, we can track the bug part here, and the crash in https://github.com/python/mypy/issues/2974

**Comment by ThomasWaldmann** [comment]
Just ran into the same issue with mypy 0.812.

Interestingly, it only happens on Github Actions / Workflows and not if I try to reproduce locally.

https://github.com/borgbackup/borg/pull/5703/checks?check_run_id=1939079054#step:9:20

Import: `from .yes import *`  # and the `yes` module contains a `yes` function
